### ISSUE TITLE
Rule proposal: `no-arguments`

### ISSUE THREAD

**Issue by axetroy** [sindresorhus/eslint-plugin-unicorn#2591]
### Description

`arguments` is a legacy and has some limitations and trap.

1. `arguments` is not a real array

```js
function example() {
  console.log(arguments.map(x => x * 2)); // ❌ TypeError
}
```

2. `arguments` doesn’t work in arrow functions

```js
function outer() {
  const arrow = () => console.log(arguments);
  arrow(4, 5, 6);
}
outer(1, 2, 3); // ❌ Outputs [1, 2, 3], not [4, 5, 6]
```

3. `arguments` doesn’t reflect default parameters

```js
function example(a = 10) {
  console.log(arguments[0]); // ❌ undefined (not 10)
}
example();
```

4. Modifying `arguments` is not allowed in `strict mode`

```js
"use strict";
function example(a) {
  arguments[0] = 100;
  console.log(a); // ❌ Still original value
}
```

### Examples

```js
// ❌
function example() {
  console.log(arguments.map(x => x * 2)); // ❌ TypeError
}

// ✅
function example(...args) {
  console.log(args.map(x => x * 2)); // ✅ [2, 4, 6]
}
```

### Proposed rule name

no-arguments

**Comment by axetroy** [comment]
My bad, I didn't search from official plugins
